internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -655,7 +655,13 @@ public ControllerResponse rollbackToBackupVersion(String storeName) {
 
   public ControllerResponse rollForwardToFutureVersion(String storeName, String regionFilter) {
     QueryParams params = newParams().add(NAME, storeName).add(REGIONS_FILTER, regionFilter);
-    return request(ControllerRoute.ROLL_FORWARD_TO_FUTURE_VERSION, params, ControllerResponse.class);
+    ControllerResponse response =
+        request(ControllerRoute.ROLL_FORWARD_TO_FUTURE_VERSION, params, ControllerResponse.class);
+    if (response.isError()) {
```

Comment:
if its error, do we need to throw exception?

Reply:
Throwing exception is the way to know that the request failed originally and also let the users know that it failed. Do you not want to throw the exception?

Reply:
Added a retry in `VeniceParentHelixAdmin#rollForwardToFutureVersion` (ref: `NUMBER_OF_RETRY_FOR_ROLL_FORWARD_CHECK`) to ease this exception out. what do you think?

Reply:
if the response is error it will be handled in the caller code, no need to throw exception here.